### ISSUE TITLE
Multiple instances support

### ISSUE THREAD

**Issue by regagain** [rugk/mastodon-simplified-federation#28]
Thank you for this great add-on!

It would be great to have support for multiple instances. I guess it would require some kind of UI that lets the user choose which one of the their instances they want to use every time there's an interaction. What do you think?

**Comment by rugk** [comment]
One important question would be:
* Is it enough if you have a popup or so, where you could select which account you want to _currently_ redirect to?
* or should it be possible to chose this each time an interaction/follow request appears?

**Comment by regagain** [comment]
> Is it enough if you have a popup or so, where you could select which account you want to currently redirect to?
I think this would be enough!

**Comment by rugk** [comment]
Okay, maybe wait for some more opinions.

Because, actually, the "choose each time" mode may be easier to implement… :laughing:

**Comment by regagain** [comment]
Both options are fine with me tbh :)

**Comment by 2rs2ts** [comment]
I would like to suggest that, if it could support firefox containers, a workflow of one account per container (which is what I do, for the most part) would prevent the need for a mandatory "ask each time" flow. Just some food for thought, though!

**Comment by rugk** [comment]
Indeed, that is a great idea. The API doc for that is https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/contextualIdentities and the permission could (and should) be optional and be requested when a setting is enabled [via the PermissionRequest lib](https://github.com/TinyWebEx/PermissionRequest/).

Note also when we have the tab, we can then retrieve the contextual (cookie store) ID of that to find out what context we currently have: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/Tab#cookiestoreid

Example: https://github.com/mdn/webextensions-examples/tree/master/contextual-identities

**Comment by testman42** [comment]
I'm happy to see progress being made about this, as #92 is an issue that I encountered as well.
